File: server/models/slider.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class Slider(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    title: str
    description: str
    button_text: str = "Shop Now"
    button_link: str = "/products"
    bg_image: str
    order: int = 0
    is_active: bool = True
    created_at: Optional[datetime] = None

    class Config:
        allow_population_by_field_name = True
        schema_extra = {
            "example": {
                "title": "Summer Sale",
                "description": "Get the best deals this summer!",
                "button_text": "Shop Now",
                "button_link": "/products",
                "bg_image": "https://example.com/image.jpg",
                "order": 1,
                "is_active": True
            }
        }

File: server/models/Slider.js
const mongoose = require('mongoose');

const sliderSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      required: true,
      trim: true,
    },
    buttonText: {
      type: String,
      default: 'Shop Now',
      trim: true,
    },
    buttonLink: {
      type: String,
      default: '/products',
      trim: true,
    },
    bgImage: {
      type: String,
      required: true,
    },
    order: {
      type: Number,
      default: 0,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
    createdAt: {
      type: Date,
      default: Date.now,
    },
    updatedAt: {
      type: Date,
      default: Date.now,
    },
  },
  { timestamps: true }
);

// Sort by order before querying
sliderSchema.query.byOrder = function () {
  return this.sort({ order: 1 });
};

module.exports = mongoose.model('Slider', sliderSchema);
